### PR TITLE
Enable multi-cluster-version testing for Jenkins and Jenkins Plugins

### PR DESCRIPTION
**Pull request by coreydaley**

### CODE REVIEW

**Comment by coreydaley** [comment]
/pj-rehearse pull-ci-openshift-jenkins-master-e2e-aws-jenkins-client-plugin-openshift-4-14

**Comment by coreydaley** [comment]
@divyansh42 Would you please take a look at this pull request and see what the difference is between what you were doing before with AWS and this one? Or maybe it was just a fluke that it passed? we should run the rehearsals again on this one once they all pass (waiting on one of the 4.14 ones still)

File: ci-operator/config/openshift/jenkins/openshift-jenkins-master.yaml [hunk 1]
```diff
@@ -57,33 +69,126 @@ tests:
...
+- as: e2e-aws-jenkins-sync-plugin-openshift-4-14
+  skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
+  steps:
+    cluster_profile: aws
+    dependencies:
+      OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:latest
+      RELEASE_IMAGE_LATEST: release:latest
+    test:
+    - ref: jenkins-sync-plugin-e2e
+    workflow: ipi-aws
```
Comment (coreydaley) [inline ci-operator/config/openshift/jenkins/openshift-jenkins-master.yaml:107]: I wonder if we should remove the `-4-14` from these and then we know they are for master branch, and then we just name the others? We will need to update this every release, but I don't' think that is a deal breaker.

Comment (coreydaley) [reply ci-operator/config/openshift/jenkins/openshift-jenkins-master.yaml]: @divyansh42 ok, i removed the `-4-14` from the master branch tests.

**Comment by coreydaley** [comment]
/pj-rehearse pull-ci-openshift-jenkins-master-e2e-aws-jenkins-sync-plugin-openshift-4-11

nfs server pod restricted must run as non root error (flake?)

**Comment by coreydaley** [comment]
@divyansh42 All tests and rehearsals passed, I updated the commit message and am running the rehearsals again

**Comment by coreydaley** [comment]
/assign @divyansh42 @mbharatk @apoorvajagtap
Please double-check my changes and make sure that I have the tests labeled correctly etc.

**Comment by coreydaley** [comment]
/hold
for review

**Comment by divyansh42** [comment]
/pj-rehearse ci/rehearse/openshift/jenkins/master/e2e-aws-openshift-4-13

**Comment by divyansh42** [comment]
/pj-rehearse pull-ci-openshift-jenkins-master-e2e-aws-openshift-4-13

**Comment by divyansh42** [comment]
/pj-rehearse pull-ci-openshift-jenkins-master-e2e-aws-jenkins-sync-plugin-openshift

**Comment by divyansh42** [comment]
> @divyansh42 Would you please take a look at this pull request and see what the difference is between what you were doing before with AWS and this one? Or maybe it was just a fluke that it passed? we should run the rehearsals again on this one once they all pass (waiting on one of the 4.14 ones still)

The only major change that I can see in your commit is, you are using
`OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:latest`
But I have used different versions in place of `latest`
I am updating my PR, let's see how it goes.

**Comment by divyansh42** [comment]
> @divyansh42 Would you please take a look at this pull request and see what the difference is between what you were doing before with AWS and this one? Or maybe it was just a fluke that it passed? we should run the rehearsals again on this one once they all pass (waiting on one of the 4.14 ones still)

The only major change that I can see in your commit is, you are using
`OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: release:latest`
But I have used different versions in place of `latest`
I am updating my PR, let's see how it goes.

**Comment by coreydaley** [comment]
/pj-rehearse pull-ci-openshift-jenkins-master-jenkins-check-plugins-supply-chain-change

**Comment by coreydaley** [comment]
/pj-rehearse pull-ci-openshift-jenkins-master-e2e-aws-openshift-4-13

**Pull request closed without merging**